tests/annotation_tests.js

```diff
@@ -18,8 +19,251 @@ VF.Test.Annotation = (function () {
       runTests('Test Justification Annotation Stem Down', Annotation.justificationStemDown);
       runTests('TabNote Annotations', Annotation.tabNotes);
     },
-
-    simple: function (options, contextBuilder) {
+    buildNotesFromJson: function(json) {
+      const rv = {
+        notes: [],
+        beamGroups: []
+      };
+      let currentBeam = [];
+      const beamNotes = () => {
+        if (currentBeam.length > 1) {
+          rv.beamGroups.push(new VF.Beam(currentBeam));
+        }
+        currentBeam = [];
+      };
+      json.notes.forEach((nn) => {
+        const keys = nn.pitches.map((pp) => pp.pitch.key);
+        const params = {
+          duration: nn.duration,
+          keys,
+          clef: nn.clef
+        };
+        const note = new VF.StaveNote(params);
+        nn.lyrics.forEach((lyric) => {
+          const vexL = new VF.Annotation(lyric.text);
+          vexL.setVerticalJustification(VF.Annotation.VerticalJustify.BOTTOM);
+          note.addAnnotation(lyric.verse, vexL);
+        });
+        if (nn.duration.indexOf('d') >= 0) {
+          note.addDotToAll();
+        }
+        if (note.ticks.value() <= 2048) {
+          currentBeam.push(note);
+          if (nn.endBeam) {
+            beamNotes();
+          }
+        } else {
+          beamNotes();
+        }
+        nn.pitches.forEach((pp, ix) => {
+          if (pp.pitch.accidental) {
+            note.addAccidental(ix, new VF.Accidental(pp.pitch.accidental));
+          }
+        });
+        rv.notes.push(note);
+      });
+      beamNotes();
+      rv.voice = new VF.Voice({
+        num_beats: json.beats.num_beats,
+        beat_value: json.beats.beat_value
+      });
+      rv.voice.addTickables(rv.notes);
+      return rv;
+    },
+    drawMusic(context, formatter, musicArray) {
+      let y = 40;
+      const voices = [];
+      musicArray.forEach((music) => {
+        formatter.joinVoices([music.voice]);
+        voices.push(music.voice);
+      });
+      const width = formatter.preCalculateMinTotalWidth(voices);
+      formatter.format(voices, width);
+      voices.forEach((voice) => {
+        const stave = new VF.Stave(10, y, width + 20);
+        y += 120;
+        stave.setContext(context).draw();
+        voice.draw(context, stave);
+      });
+      musicArray.forEach((music) => {
+        music.beamGroups.forEach((beam) => {
+          beam.setContext(context).draw();
+        });
+      });
+    },
+    lyrics: function(options) {
```

Comment:
This seems pretty verbose, why not use EasyScore instead (which is what I use for any new tests.)

Reply:
Sure, is there a way to add Annotations?  Or should I just go `system.part.voices[0].tickables[i].addAnnotation...`.

Reply:
Yes, you can (by giving IDs to the notes):

```
      system = makeSystem(150);
      system.addStave({ voices: [voice(notes('D5/q[id="m2a"], G4, G4'))] });
      system.addStave({ voices: [voice(notes('B3/h.', { clef: 'bass' }))] });
      system.addConnector('singleRight');

      id('m2a').addModifier(0, vf.Articulation({ type: 'a.', position: 'above' }));
```

See `tests/bach_tests.js` for a full example.